### PR TITLE
Increases Coverage on Stake Manager

### LINKED ISSUE #258: Increase test coverage on stakeManager

### PR DESCRIPTION
**Pull request by GauravJain9**
Fixes #258 
Added test cases to increase coverage on stake manager file

### CODE REVIEW

**Comment by GauravJain9** [comment]
Tests are all running locally don't know why it's not passing here 
*This warning (unhandled promise rejection is coming locally also when I am running test cases)

**Comment by abhishekvispute** [comment]
> Tests are all running locally don't know why it's not passing here
> *This warning (unhandled promise rejection is coming locally also when I am running test cases)

Its because of missing await, please add those wherever needed

File: test/StakeManager.js [hunk 1]
```diff
@@ -331,6 +403,32 @@ describe('StakeManager', function () {
       assertBNEqual(staker.stake, stake, 'Stake should not change');
     });
 
+    it('Staker should not be able to withdraw if committed during lock period despite not revealing', async function () {
+      let epoch = await getEpoch();
+      const amount = tokenAmount('10000');
+      const stakerId = stakeManager.stakerIds(signers[2].address);
```
Comment (abhishekvispute) [inline test/StakeManager.js:409]: like this

Comment (GauravJain9) [reply test/StakeManager.js:409]: I checked the file and added await in 2 lines and checked file rest of the await is there where it should be, but the warning is still showing particularly only for this test case https://github.com/razor-network/contracts/pull/312/files#diff-0ea851f7ce30fd6fc4b4e3c5babae84457d03368116d3c1d4212c2b04cb2a575R559
await is there and also after adding await at two more places the same warning for same particular test case is still there don't know exactly why is this happening.

Comment (abhishekvispute) [reply test/StakeManager.js:409]: hey @GauravJain9  is this solved ?

Comment (GauravJain9) [reply test/StakeManager.js:409]: yes it is solved @abhishekvispute will update

Comment (GauravJain9) [reply test/StakeManager.js]: Done and updated @abhishekvispute

**Review by abhishekvispute — DISMISSED** [review]

**Review by SamAg19 — DISMISSED** [review]

File: test/StakeManager.js [hunk 2]
```diff
@@ -1000,5 +1193,21 @@ describe('StakeManager', function () {
...
     });
+
+    it('Staker should not be able to withdraw if his stake is zero', async function () {
+      let epoch = await getEpoch();
+      const stakerIdacc3 = await stakeManager.stakerIds(signers[3].address);
+      await stakeManager.connect(signers[3]).unstake(epoch, stakerIdacc3, tokenAmount('1000'));
+      for (let i = 0; i < WITHDRAW_LOCK_PERIOD - 1; i++) {
+        await mineToNextEpoch();
+      }
+      await mineToNextEpoch();
```
Comment (hrishikeshio) [inline test/StakeManager.js:1204]: why is this here? just remove -1 is the loop above.

File: test/StakeManager.js [hunk 3]
```diff
@@ -872,6 +1021,50 @@ describe('StakeManager', function () {
...
+    });
+
+    it('Staker should not be able to withdraw if the stakemanager contract is out of funds', async function () {
+      let epoch = await getEpoch();
+      const stakerIdacc3 = await stakeManager.stakerIds(signers[3].address);
+      await stakeManager.connect(signers[3]).unstake(epoch, stakerIdacc3, tokenAmount('1000'));
+      for (let i = 0; i < WITHDRAW_LOCK_PERIOD - 1; i++) {
+        await mineToNextEpoch();
+      }
+      await mineToNextEpoch();
```
Comment (hrishikeshio) [inline test/StakeManager.js:1035]: same

**Review by abhishekvispute — APPROVED** [review]

File: test/StakeManager.js (hunk 3 above)
Comment (GauravJain9) [reply test/StakeManager.js]: Done @hrishikeshio

File: test/StakeManager.js (hunk 2 above)
Comment (GauravJain9) [reply test/StakeManager.js]: ok

**Review by hrishikeshio — APPROVED** [review]
